src/protocol/sort/generate_sort_permutation.rs

```diff
@@ -147,4 +201,51 @@ mod tests {
 
         Ok(())
     }
+
+    #[tokio::test]
+    pub async fn test_shuffle_and_reveal_permutation() {
+        const BATCHSIZE: u32 = 25;
+
+        let mut rng = rand::thread_rng();
+
+        for _ in 0..10 {
```

Comment:
Do you really need to do this 10 times?  With 24 items, the odds of the order being the same is $\frac{1}{25!}$, which is small enough.